### ISSUE TITLE
Design doc review #1

### ISSUE THREAD

**Issue by aracpyon** [nickdraper8/RubyReel#1]
Hi Nick! Looking great so far! Here are my suggestions

**MVP**
- [x] add update and destroy in video uploads

**Schema**
- [x] add `email` to `User` table
- [x] You don't need `file_path` column for `Videos` (Can be done in the frontend and we are going to use AWS to handle uploading)
- [x] Get rid of `views` column from `Videos`. Views should be a separate table. You might want to implement this feature later down the line as a bonus. We need to think about how you are going to count the views as well.

**Comment by nickdraper8** [comment]
Hi Ara,

I tried to address all your suggestions, and I feel as though I was able to make some beneficial changes.

I did however want your opinion on my change involving the views. I made it's own table, and I am going to make it a table that has relations of video_id to ip_address. Is this the right way to go about counting video views? Let me know what you think. 

Also, for AWS, I am assuming I will be able to retrieve video information just with the video_id primary key? Will we learn more about how to use AWS or will that have to be learned independently?

Thanks!

**Comment by aracpyon** [comment]
Hey Nick! 

Great question. Views is a bonus feature because it is not required. Also it's different based on how you are going to calculate the views.  It depends on how you are going to calculate the views. Right now Youtube is counting the views based on the run-time of ads on the video. 
I was thinking to calculate the views based on the number of clicks by logged-in users so was suggesting to make a `View` table that contains `viewer_id` and `video_id`. However, based on your plans, things can be different. :) Since this is a bonus feature, I promise we will get back to this discussion after finishing all the features. 

^^**_update_**: Actually even non-logged-in users can play the videos, so I guess you don't need a separate table in this case. If you are just counting the number of plays through query/rendering the component or a special `setTimeout` to add view only after playing the video for a while, just having a `view` column on the table sounds good.  But we are open to other ways too. Let's talk more when we reach to this feature! :) 

For your second question, yes we are going to provide a series of videos and documents on AWS for you to able to implement it for your project (they are already on Open a/A under actually! Check fullstack > resources). AWS is a cloud web service provided by Amazon. When you upload the video/photos, those medias will be saved on AWS cloud and you are accessing them through the url given by AWS. 

I have another suggestion. I just saw your states and I found that you are having a slice of state to `dislikeIds` on comments and videos. That is alright,  but when you send back the data to frontend from jbuilder, just to make sure to filter the information into two slices(likes and dislikes) from your `Likes` table since you are sharing the table for both likes and dislikes. 

Let me know if you have any other questions!
